src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -103,6 +103,15 @@ private static void HandleEndDictionary(JsonSerializerOptions options, ref Utf8J
 
             if (state.Current.IsDictionaryProperty)
             {
+                // Handle special case of DataExtensionProperty where we just added a dictionary element to the extension property.
+                // Since the JSON value is not a dictionary element (just a normal property value) a JsonTokenType.EndObject
+                // encountered here is from the outer object so forward to HandleEndObject().
+                if (state.Current.JsonClassInfo.DataExtensionProperty == state.Current.JsonPropertyInfo)
+                {
+                    HandleEndObject( ref reader, ref state);
+                    return;
```

Comment:
nit: Can this be an if/else so that we can remove this `return` statement in the middle of the method (and let the code just fall through to the end). Currently its the only one and its within a nested condition.

Reply:
Yep. Thanks